### ISSUE TITLE
molsurf: set radii

### ISSUE THREAD

**Issue by hainm** [Amber-MD/cpptraj#102]
from the doc

`This routine currently requires radius information to be present in the topology file`

Should we add SetRadii for Topology so we can use pdb file?

**Comment by drroe** [comment]
Fixed via #132

**Comment by hainm** [comment]
@drroe what's about arbitrary radii? For example. somebody come up with new radii set and want to test molsurf. Do they need to use ParmEd to update radii, save to parm7 and let cpptraj read the radii?

**Comment by hainm** [comment]
just for example with trpcage pdb file (1L2Y). with your new update, I got

```bash
Warning: Setting topology box information from trajectory.
  [molsurf out test.dat]
    MOLSURF: '*' Probe Radius= 1.400 Ang.
    Using GB radii.
---------- RUN BEGIN -------------------------------------------------

PARAMETER FILES:
 0: 1L2Y.pdb, 304 atoms, 20 res, box: Orthogonal, 1 mol

INPUT TRAJECTORIES:
 0: '1L2Y.pdb' is a PDB file, Parm 1L2Y.pdb (Orthogonal box) (reading 38 of 38)
  Coordinate processing will occur on 38 frames.
TIME: Run Initialization took 0.0000 seconds.

BEGIN TRAJECTORY PROCESSING:
.....................................................
ACTION SETUP FOR PARM '1L2Y.pdb' (1 actions):
  0: [molsurf out test.dat]
    Calculating surface area for 304 atoms.
Warning: Atom 'ASN_1@N' has 0.0 GB radius.
Warning: Atom 'ASN_1@CA' has 0.0 GB radius.
Warning: Atom 'ASN_1@C' has 0.0 GB radius.
Warning: Atom 'ASN_1@O' has 0.0 GB radius.
Warning: Atom 'ASN_1@CB' has 0.0 GB radius.
Warning: Atom 'ASN_1@CG' has 0.0 GB radius.
Warning: Atom 'ASN_1@OD1' has 0.0 GB radius.
```

**Comment by hainm** [comment]
### Segmentation fault

```bash
parm 1L2Y.pdb
trajin 1L2Y.pdb
molsurf out test.dat radii vdw
```

```bash

CPPTRAJ: Trajectory Analysis. V16.00b OpenMP
    ___  ___  ___  ___
     | \/ | \/ | \/ |
    _|_/\_|_/\_|_/\_|_

| Date/time: 10/15/15 17:30:00
| Available memory: 6163.49 MB

INPUT: Reading Input from file trpcage.molsurf_pdb.in
  [parm 1L2Y.pdb]
        Reading '1L2Y.pdb' as PDB File
        Determining bond info from distances.
Warning: 1L2Y.pdb: Determining default bond distances from element types.
  [trajin 1L2Y.pdb]
        Reading '1L2Y.pdb' as PDB
        Read CRYST1 info from PDB: a=1 b=1 c=1 alpha=90 beta=90 gamma=90
Warning: PDB cell lengths are all 1.0 Ang.; this usually indicates an invalid box.
Warning: Trajectory box type is 'Orthogonal' but topology box type is 'None'.
Warning: Setting topology box information from trajectory.
  [molsurf out test.dat radii vdw]
    MOLSURF: '*' Probe Radius= 1.400 Ang.
        Using vdW radii.
---------- RUN BEGIN -------------------------------------------------

PARAMETER FILES:
 0: 1L2Y.pdb, 304 atoms, 20 res, box: Orthogonal, 1 mol

INPUT TRAJECTORIES:
 0: '1L2Y.pdb' is a PDB file, Parm 1L2Y.pdb (Orthogonal box) (reading 38 of 38)
  Coordinate processing will occur on 38 frames.
TIME: Run Initialization took 0.0000 seconds.

BEGIN TRAJECTORY PROCESSING:
.....................................................
ACTION SETUP FOR PARM '1L2Y.pdb' (1 actions):
  0: [molsurf out test.dat radii vdw]
        Calculating surface area for 304 atoms.
Segmentation fault
```

**Comment by drroe** [comment]
> what's about arbitrary radii? For example. somebody come up with new radii set and want to test molsurf. Do they need to use ParmEd to update radii, save to parm7 and let cpptraj read the radii?

They can just use a PQR file, which is a modified PDB file where occupancy/B-factor columns are replaced by charge/radius, e.g. `parm custom.pqr pqr`.

**Comment by drroe** [comment]
Segfault should be fixed by #135

**Comment by hainm** [comment]
close this issue since we have #133
